Development/nmos/control_protocol_state.cpp

```diff
@@ -178,9 +178,30 @@ namespace nmos
                     return get_datatype(resources, resource, arguments, is_deprecated, get_control_protocol_datatype_descriptor, gate);
                 };
             }
+            nmos::experimental::control_protocol_method_handler make_nc_get_properties_by_path(experimental::control_protocol_method_handler control_protocol_method_handler)
+            {
+                return [control_protocol_method_handler](nmos::resources& resources, const nmos::resource& resource, const web::json::value& arguments, bool is_deprecated, slog::base_gate& gate)
+                {
+                    return get_properties_by_path(resources, resource, arguments, is_deprecated, control_protocol_method_handler, gate);
+                };
+            }
+            nmos::experimental::control_protocol_method_handler make_nc_validate_set_properties_by_path(experimental::control_protocol_method_handler control_protocol_method_handler)
+            {
+                return [control_protocol_method_handler](nmos::resources& resources, const nmos::resource& resource, const web::json::value& arguments, bool is_deprecated, slog::base_gate& gate)
+                    {
+                        return validate_set_properties_by_path(resources, resource, arguments, is_deprecated, control_protocol_method_handler, gate);
+                    };
+            }
+            nmos::experimental::control_protocol_method_handler make_nc_set_properties_by_path(experimental::control_protocol_method_handler control_protocol_method_handler)
+            {
+                return [control_protocol_method_handler](nmos::resources& resources, const nmos::resource& resource, const web::json::value& arguments, bool is_deprecated, slog::base_gate& gate)
+                    {
+                        return set_properties_by_path(resources, resource, arguments, is_deprecated, control_protocol_method_handler, gate);
+                    };
+            }
```

Comment:
I think it would be better to define different callback handlers for get_properties_by_path, validate_set_properties_by_path, and set_properties_by_path rather than using the standard experimental::control_protocol_method_handler and put them in the control_protocol_handlers.h
e.g.
typedef std::function<web::json::value(const nmos::resource& resource, const web::json::value& arguments)> get_properties_by_path_handler

Reply:
As above, I will add typedefs.